src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTag.cs

```diff
@@ -0,0 +1,153 @@
+        /// </summary>
+        public override GraphicsResult GetGraphics(IWpfTextView view, Geometry bounds, TextFormattingRunProperties format)
+        {
+            var block = new TextBlock
+            {
+                FontFamily = format.Typeface.FontFamily,
+                FontSize = 0.75 * format.FontRenderingEmSize,
+                FontStyle = FontStyles.Normal,
+                Foreground = format.ForegroundBrush,
+                Padding = new Thickness(left: 2, top: 0, right: 2, bottom: 0),
+            };
+
+            var id = new Run(_diagnostic.Id);
+            var link = new Hyperlink(id)
+            {
+                NavigateUri = new Uri(_diagnostic.HelpLink)
+            };
+
+            var moniker = GetMoniker();
+            if (moniker is not null)
+            {
+                var image = new CrispImage
+                {
+                    Moniker = (ImageMoniker)moniker,
+                    RenderSize = new Size(16, 16),
+                    Margin = new Thickness(1, 0, 5, 0)
+                };
+
+                var statusImage = new InlineUIContainer(image);
+                block.Inlines.Add(statusImage);
+            }
+
+            link.RequestNavigate += HandleRequestNavigate;
+            block.Inlines.Add(link);
+            block.Inlines.Add(": " + _diagnostic.Message);
+            block.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            block.Arrange(new Rect(block.DesiredSize));
+
+            var border = new Border
+            {
+                BorderBrush = format.BackgroundBrush,
+                BorderThickness = new Thickness(1),
+                Background = Brushes.Transparent,
+                Child = block,
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(10, top: 0, 0, bottom: 0),
+                Padding = new Thickness(1)
+            };
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            // Need to set these properties to avoid unnecessary reformatting because some dependancy properties
+            // affect layout.
+            // TODO: Not sure if these are needed anymore since the errors are not intratextadornment tags
+            TextOptions.SetTextFormattingMode(border, TextOptions.GetTextFormattingMode(view.VisualElement));
+            TextOptions.SetTextHintingMode(border, TextOptions.GetTextHintingMode(view.VisualElement));
+            TextOptions.SetTextRenderingMode(border, TextOptions.GetTextRenderingMode(view.VisualElement));
+
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+
+            return new GraphicsResult(border,
+                    () => view.ViewportWidthChanged -= ViewportWidthChangedHandler);
```

Comment:
1. named parameter for lambda.
2. do you also need to disconnect: link.RequestNavigate += HandleRequestNavigate;